### PR TITLE
[OSPP]The proposal of unknow task recognition algorithm reproduction

### PR DESCRIPTION
**Pull request by Frank-lilinjie**
**What type of PR is this?**
/kind feature

**What this PR does:**
This PR is a proposal that project aims to reproduce the CVPR2021 paper "Learning placeholders for open-set recognition" in the Semantic Segmentation dataset. This paper proposes placeholders that imitate the emergence of new classes, thus helping to transform closed-set training into open-set training.

### CODE REVIEW

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
+# Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna
+Traditional machine learning performs test-set inference by training known samples to which its knowledge is limited. The model which is trained by traditional machine learning cannot be effectively recognized the unknown samples from new classes and they will be processed as known samples. Therefore, the recognition and processing of unknown samples or unknown tasks will become the main research direction of artificial intelligence in the future. This project aims to reproduce the CVPR2021 paper "Learning placeholders for open-set recognition" in the Semantic Segmentation dataset. This paper proposes placeholders that imitate the emergence of new classes, thus helping to transform closed-set training into open-set training.
+
+
+
+## Goals
+1. The reproduction should be completed based on the Cityscapes Semantic Segmentation dataset.
```
Comment (MooreZheng) [inline docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md:7]: Good to know the Cityscapes dataset in scope. And how about the Synthia Dataset? Scenario description of the Cityscape dataset is also needed, e.g., [PCB-AoI](https://ianvs.readthedocs.io/en/latest/proposals/scenarios/industrial-defect-detection/pcb-aoi.html).

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+2. The reproduced code is successfully merged into the lifelong learning module of Sedna.
```
Comment (MooreZheng) [inline docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md:8]: Well, not Sedna here and should be Ianvs. Besides, the algorithm description is also needed, e.g., [Incremental learning: BasicIL-FPN](https://ianvs.readthedocs.io/en/latest/proposals/algorithms/incremental-learning/basicIL-fpn.html).

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+3. The recognition accuracy (e.g. f1_score) of unknown classes is greater than 0.9.
```
Comment (MooreZheng) [inline docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md:9]: Test report of this algorithm is also needed, e.g., [Testing incremental learning in industrial defect detection](https://ianvs.readthedocs.io/en/latest/proposals/test-reports/testing-incremental-learning-in-industrial-defect-detection-with-pcb-aoi.html).

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+**PROSER algorithm training process**
+
+![](images/process.png)
+
+### Embedded in Ianvs
+
+![ ](images/ianvs_arch.png)
+
+After training, the unknown task algorithm model is placed in the Lifelong Learning Paradigm section of the Test Case Controller module as one of the algorithms for unknown task identification in the lifelong learning paradigm.
```
Comment (MooreZheng) [inline docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md:120]: Good to learn that. It is true that the algorithm should put in the lifelong learning lib in the test case controller. But currently in this proposal the ianvs lifelong learning procedure is not clear. How about put the ianvs lifelong modules in a figure?

**Comment by MooreZheng** [comment]
The proposal is overall fine with minor revisison needed. Remind The proposed lifelong learning Benchmarking files should match the ianvs directories. Suggestion:

ianvs
|- example
    |- aaa (named as the used dataset)
        |- lifelong_learning_bench
            |- testalgorithm
                |- mynet.yaml (named as the used algorithm)
            |- testenv
                |- testenv.yaml
                |- mAP.py (named as the used metric)
            |- benchmarkingjob.yaml

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md [hunk 5]
```diff
@@ -0,0 +1,254 @@
...
+   - dataset:
+     values:
+       - "dataset_test"
+```
+
+
+
+## 4 datasets
+
+In this project, the image source is used as the classification criterion, and RGB images in three datasets are selected as samples, among which, the real camera dataset and simulator dataset are selected for the known task dataset, and the animation dataset of Akiyama open source is selected for the unknown task dataset. The sample dataset and dataset structure are shown below.
```
Comment (MooreZheng) [inline docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:91]: To fit a real world setting, it is recommended to set the real camera dataset as unknown.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md [hunk 6]
```diff
@@ -0,0 +1,254 @@
...
+
+In this project, the algorithm is packaged as a python callable module and embedded in the lib library of Ianvs' lifelong learning testing system. The algorithm developer does not need to develop additional algorithms for unknown task recognition and can directly test the performance of the currently developed algorithms in combination with the dataset and testing environment provided by Ianvs. At the same time, Ianvs provides local and cloud-based algorithm performance rankings for developers to facilitate the exchange of lifelong machine learning researchers and thus promote the development of the lifelong learning research field.
+
+
+
+## 2 Goals
+
+1. Reproducing the algorithm from the CVPR2021 paper "Learning placeholders for open-set recognition".
+2. Achieving an accuracy of 0.9 for unknown task recognition.
+3. Making unknown category recognition modules according to Ianvs architecture and contextual relationship to store in the lib library, so that the algorithm can be used on demand.
```
Comment (MooreZheng) [inline docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:29]: In the example, except algorithm-wise procedures like reproduction, tuning and ianvs lib development, the goals should also include

 documentation:
1. scenario description of the autopilot dataset, e.g., [PCB-AoI](https://ianvs.readthedocs.io/en/latest/proposals/scenarios/industrial-defect-detection/pcb-aoi.html)
2. a multi-task joint inference algorithm and its description, e.g., [Incremental learning: BasicIL-FPN](https://ianvs.readthedocs.io/en/latest/proposals/algorithms/incremental-learning/basicIL-fpn.html)
3. test report of this algorithm, e.g., [Testing incremental learning in industrial defect detection](https://ianvs.readthedocs.io/en/latest/proposals/test-reports/testing-incremental-learning-in-industrial-defect-detection-with-pcb-aoi.html)

Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:29]: The comment is not yet revised so far.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md (hunk 5 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:91]: It seems that this comment is resolved.

**Comment by luosiqi** [comment]
This commit should not be included codes but proposal.md only.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 1 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Please follow up this comment.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 2 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Please follow up this comment.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
+# Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs
+Traditional machine learning performs test-set inference by training known samples to which its knowledge is limited. The model which is trained by traditional machine learning cannot be effectively recognized the unknown samples from new classes and they will be processed as known samples. Therefore, the recognition and processing of unknown samples or unknown tasks will become the main research direction of artificial intelligence in the future. This project aims to reproduce the CVPR2021 paper "Learning placeholders for open-set recognition" in the Semantic Segmentation dataset. This paper proposes placeholders that imitate the emergence of new classes, thus helping to transform closed-set training into open-set training.
+
+
+
+## Goals
```
[the remark this answers is not in the document]
Comment (luosiqi) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Not yet modified.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 3 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Please follow up this comment.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 4 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: We see the ianvs arch but the relationship between the proposal and ianvs can be more straightforward. My suggestion is to add a figure showing the proposal module in the overall lifelong learning flow to see how it works.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 1 above)
Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: ok

Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Received and completed.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 2 above)
Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Received and completed.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 3 above)
Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Received. After the embedded work is completed, the test shall be conducted and ended when the test report is obtained.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md (hunk 5 above)
Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:91]: Yep, this comment has been modified in the experiment, but this version of the proposal has not been modified.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md (hunk 6 above)
Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:29]: Description of the dataset and an unknow task recognition algorithm and its description are completed.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 1 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: ok

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 2 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: ok

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 3 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: still need follow up

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 4 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: still need to follow up. highlight the proposed module in overall flow.

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md (hunk 6 above)
Comment (MooreZheng) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Ianvs.md:29]: please add these items into goals

**Comment by MooreZheng** [comment]
The proposal is overall good but
1. pls leave only one markdown in the proposal directory. other files can be put in sub-directory
2. revised the mentioned unresolved comment

File: docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md (hunk 4 above)
Comment (Frank-lilinjie) [reply docs/proposals/algorithms/lifelong-learning/Unknown Task Recognition Algorithm Reproduction based on Lifelong Learning of Sedna.md]: Completed~

**Comment by MooreZheng** [comment]
This PR is outdated and thus reverted. Please refer to the updated PR https://github.com/kubeedge/ianvs/pull/41

**Pull request merged**